core/src/main/java/greencity/service/impl/UserServiceImpl.java

```diff
@@ -722,4 +726,25 @@ public void addNewFriend(Long userId, Long friendId) {
     public void updateUserLastActivityTime(Long userId, Date userLastActivityTime) {
         userRepo.updateUserLastActivityTime(userId, userLastActivityTime);
     }
+
+    /**
+     * The method checks by id if a {@link User} is online.
+     *
+     * @param userId {@link Long}
+     * @return {@link Boolean}.
+     * @author Yurii Zhurakovskyi
+     */
+    @Override
+    public Boolean checkIfTheUserIsOnline(Long userId) {
+        Date userLastActivityTime = userRepo.findLastActivityTimeById(userId)
+                .orElseThrow(() -> new UserLastActivityTimeNotFoundException(
+                        USER_LAST_ACTIVITY_TIME_NOT_FOUND + userId));
+        Date currentTime = new Date();
+        long result = currentTime.getTime() - userLastActivityTime.getTime();
```

Comment:
Here is one more reason to have a separate table with not nullable column - this calculation should be part of the query.